qstnrs/questionnaire/utils.py

```diff
@@ -0,0 +1,65 @@
+from questionnaire.models import Answer
+
+
+def get_score_for(user_choices):
+    """Calculate the user's total score based on selected answers."""
+    return sum(list(Answer.objects.filter(id__in=user_choices).values_list(
+        'answer_score', flat=True)))
+
+
+def get_max_score_for(questionnaire_id):
+    """Get the maximum score the user could have obtained if he/she would have
+     chosen only the answers with a positive score"""
+    score_list = Answer.objects.filter(
+        question__page__questionnaire_id=questionnaire_id,
+        answer_score__gt=0).values_list('answer_score', flat=True)
+    return sum(score_list)
+
+
+def select_optimal_answers(user_choices, unselected_choices, better):
+    """Select all the answers the user could have selected for a better or
+    worse score.--add order-to-doc"""
+    user_score = get_score_for(user_choices)
+
+    possible_answers = []
+    accum = user_score
+    for choice in unselected_choices:
+        # Only consider answers from different pages or from the same page
+        # and same question
+        if on_same_page(choice, possible_answers):
+            continue
+        if better and accum <= user_score:
+            possible_answers.append(choice)
+            accum += choice.answer_score
+        elif not better and accum >= user_score:
+            possible_answers.append(choice)
+            accum += choice.answer_score
+    return possible_answers
+
+
+def on_same_page(answer, possible_choices):
```

Comment:
You should move this function before the function where it's used. It makes everything much more clearer for the one that's reading the code.